src/main.cpp

```diff
@@ -2540,16 +2631,35 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
 
         if (!tx.IsCoinBase())
         {
-            if (!view.HaveInputs(tx))
+            if (!viewTempCache.HaveInputs(tx)) {
+                // ** In the case of IBD we have to check if the height has changed while we're checking inputs because this thread can continue for a
+                // short while after it has been interrupted and meanwhile the view of the UTXO could have changed after another thread has updated the tip.
+                // ** Additionally and more importantly in the event of a big block DDOS attack, by checking the chain height here, we ensure that
+                //    the thread that is validating the big block will immediately exit and finish up (while still keeping the block on disk in the
+                //    case of a reorg) as soon as the first block makes it through and wins the validation race.
```

Comment:
@dgenr8 I must say though I don't like the idea of having to put some duplicate code in that loop that is not necessary, I think it's fine the way it is, it's not vital that the thread self terminates at that point, if it does then fine, but it will get terminated either from the thread that wins the race or further down stream; at the last check just before it would update the UTXO, if it makes it that far it will finally self terminate.